### PR TITLE
Add diary button functionalities and update CSS Styles

### PR DESCRIPTION
**Pull request by Gracekangdh**
## changes
- Added functionalities : create new diary and save them in local storage
- Implemented Delete button 
- Added Edit, Save buttons on Detail page
- Updated CSS : improved button styles, and consistency, adjusted padding, margins 
- Enabled DarkMode

### CODE REVIEW

**Review by ysyoo11 — CHANGES_REQUESTED** [review]
- Remove unnecessary comments and `console.log` functions
- Fix `handleSave` in DiaryDetail
- Fix the PR title and description to elaborate what you're working on

**Review by ysyoo11 — CHANGES_REQUESTED** [review]
> * Remove unnecessary comments and `console.log` functions
> * Fix `handleSave` in DiaryDetail
> * Fix the PR title and description to elaborate what you're working on

File: src/components/Dashboard/Dashboard.jsx [hunk 1]
```diff
@@ -15,46 +16,21 @@ export default function Dashboard() {
...
-  // const storedDiaries = localStorage.getItem("diaries");
-  // if (storedDiaries) { // truthy : [] <-> null undefined
-  //   setDiaries(JSON.parse(storedDiaries));
-  // } -> (한번만) 저장되어있는 것을 GET! 상태를 업데이트하고 useEffect...
-  // 내가 쓸 로직이 함수 안에 혹은 밖에 있어야 하는가 확인!
-
-  useEffect(() => {
-    localStorage.setItem("diaries", JSON.stringify(diaries));
-  }, [diaries]);
+  useEffect(() => {}, [diaries]);
```
Comment (ysyoo11) [inline src/components/Dashboard/Dashboard.jsx]: You don't need this useEffect

File: src/pages/DiaryCard/DiaryCard.jsx [hunk 2]
```diff
@@ -1,29 +1,17 @@
+import { useNavigate } from "react-router-dom";
 import styles from "../DiaryCard/DiaryCard.module.css";
-import { MdModeEdit, MdDelete } from "react-icons/md";
 
-export default function DiaryCard({ diary, onDelete, onEdit }) {
+export default function DiaryCard({ diary, onDelete }) {
```
Comment (ysyoo11) [inline src/pages/DiaryCard/DiaryCard.jsx]: You don't need `onDelete` prop any more

File: src/components/Dashboard/Dashboard.jsx [hunk 3]
```diff
@@ -15,46 +16,21 @@ export default function Dashboard() {
...
 
   return (
     <div className={styles.diaryList}>
       {showNewDiaryForm && (
         <NewDiaryForm onAdd={handleAdd} className={styles.newDiaryForm} />
       )}
-      <DiaryList diaries={diaries} onDelete={handleDelete} onSave={onSave} />
+      <DiaryList diaries={diaries} onDelete={deleteDiary} />
```
Comment (ysyoo11) [inline src/components/Dashboard/Dashboard.jsx]: no need to pass `deleteDiary`

File: src/pages/DiaryList/DiaryList.jsx [hunk 4]
```diff
@@ -1,37 +1,13 @@
...
-  const handleSave = (updatedDiary) => {
-    onSave(updatedDiary);
-    setEditingDiaryId(null);
-  };
-
-  const cancelEditMode = () => {
-    setEditingDiaryId(null);
-  };
 
+export default function DiaryList({ diaries, onDelete }) {
```
Comment (ysyoo11) [inline src/pages/DiaryList/DiaryList.jsx]: you don't need `onDelete` prop

File: src/pages/DiaryDetail/DiaryDetail.jsx [hunk 5]
```diff
@@ -1,17 +1,94 @@
...
+
+  useEffect(() => {
+    const foundDiary = diaries.find((diary) => diary.id === id);
+
+    if (foundDiary) {
+      setDiary(foundDiary);
+      setNewDiary({ title: foundDiary.title, text: foundDiary.text });
+    } else {
+      setDiary(null);
+    }
```
Comment (ysyoo11) [inline src/pages/DiaryDetail/DiaryDetail.jsx]: Re-consider UX when diary is not found

**Review by ysyoo11 — CHANGES_REQUESTED** [review]

File: src/pages/DiaryCard/DiaryCard.jsx (hunk 2 above)
Comment (Gracekangdh) [reply src/pages/DiaryCard/DiaryCard.jsx]: removed the unnecessary 'onDelete' prop

File: src/components/Dashboard/Dashboard.jsx (hunk 1 above)
Comment (Gracekangdh) [reply src/components/Dashboard/Dashboard.jsx]: deleted useEffect

File: src/pages/DiaryList/DiaryList.jsx (hunk 4 above)
Comment (Gracekangdh) [reply src/pages/DiaryList/DiaryList.jsx]: deleted onDeleted Prop

File: src/components/Dashboard/Dashboard.jsx (hunk 3 above)
Comment (Gracekangdh) [reply src/components/Dashboard/Dashboard.jsx]: deleted onDelete and deleteDiary

File: src/pages/DiaryDetail/DiaryDetail.jsx (hunk 5 above)
Comment (Gracekangdh) [reply src/pages/DiaryDetail/DiaryDetail.jsx]: using notFound component making UX clear

File: src/components/Dashboard/Dashboard.jsx (hunk 1 above)
Comment (ysyoo11) [reply src/components/Dashboard/Dashboard.jsx]: resolved by https://gist.github.com/qoomon/5dfcdf8eec66a051ecd85625518cfd13

Comment (ysyoo11) [reply src/components/Dashboard/Dashboard.jsx]: resolved by [4144c22](https://github.com/Gracekangdh/DiaryApp/pull/1/commits/4144c229ed60f9ba0564b706a9f99ec9040ade25)

File: src/pages/DiaryCard/DiaryCard.jsx (hunk 2 above)
Comment (Gracekangdh) [reply src/pages/DiaryCard/DiaryCard.jsx]: resolved by [eab60fc](https://github.com/Gracekangdh/DiaryApp/pull/1/commits/eab60fc9b08f005d34e90e596d6061603365c022)

File: src/components/Dashboard/Dashboard.jsx (hunk 3 above)
Comment (Gracekangdh) [reply src/components/Dashboard/Dashboard.jsx]: resolved https://github.com/Gracekangdh/DiaryApp/pull/1/commits/75209445563bf42338df5bbeae5ee06b9f6ffc35

File: src/pages/DiaryList/DiaryList.jsx (hunk 4 above)
Comment (Gracekangdh) [reply src/pages/DiaryList/DiaryList.jsx]: resolved by https://github.com/Gracekangdh/DiaryApp/pull/1/commits/2120610c15e9fbd68d98ef37b1ba1811a24f50f6

File: src/pages/DiaryDetail/DiaryDetail.jsx (hunk 5 above)
Comment (Gracekangdh) [reply src/pages/DiaryDetail/DiaryDetail.jsx]: resolved by[ ](https://github.com/Gracekangdh/DiaryApp/pull/1/commits/2c63a668c98a83bb9475c05ffa66b1e9032810fd)

File: src/pages/DiaryDetail/DiaryDetail.jsx [hunk 6]
```diff
@@ -1,17 +1,96 @@
...
+  };
+
+  useEffect(() => {
+    const foundDiary = diaries.find((diary) => diary.id === id);
+
+    if (foundDiary) {
+      setDiary(foundDiary);
+      setNewDiary({ title: foundDiary.title, text: foundDiary.text });
+    } else {
+      setDiary(false);
```
Comment (ysyoo11) [inline src/pages/DiaryDetail/DiaryDetail.jsx]: Using both `false` and `null` on a single state is chaotic..
Why not just throw an error? React will handle navigating to `NotFound` automatically

File: src/pages/DiaryDetail/DiaryDetail.jsx (hunk 5 above)
Comment (Gracekangdh) [reply src/pages/DiaryDetail/DiaryDetail.jsx]: resolved by https://github.com/Gracekangdh/DiaryApp/pull/1/commits/2c63a668c98a83bb9475c05ffa66b1e9032810fd

**Review by ysyoo11 — CHANGES_REQUESTED** [review]

File: src/pages/DiaryDetail/DiaryDetail.jsx (hunk 6 above)
Comment (Gracekangdh) [reply src/pages/DiaryDetail/DiaryDetail.jsx]: resolved by[ ](https://github.com/Gracekangdh/DiaryApp/pull/1/commits/0299d983cb308487853f1843f29103962ba14f4e)

Comment (Gracekangdh) [reply src/pages/DiaryDetail/DiaryDetail.jsx]: resolved by https://github.com/Gracekangdh/DiaryApp/pull/1/commits/0299d983cb308487853f1843f29103962ba14f4e

**Review by ysyoo11 — APPROVED** [review]
LGTM

**Pull request merged**